wfapi/error.go

```diff
@@ -382,3 +386,18 @@ func ErrorCatalogName(name string, reason string) Error {
 		},
 	}
 }
+
+// ErrorFileAlreadyExists is used when a file already exists
+//
+// Errors:
+//
+//    - warpforge-error-already-exists --
+func ErrorFileAlreadyExists(path string) Error {
+	return &ErrorVal{
+		CodeString: errCodeAlreadyExists,
+		Message:    fmt.Sprintf("file already exists at path: %q", path),
+		Details: [][2]string{
+			{"path", path},
```

Comment:
fwiw, this is all going to get better when we switch over to using the [go-serum](https://github.com/serum-errors/go-serum) package instead of our hand-rolled things here.  It allows attaching causes easily with either `%w` in `serum.Errorf` constructor, or `serum.WithCause` for the fuller `serum.Error` constructor.  So this could probably disappear into a puff of simply attaching the file IO error as a cause.